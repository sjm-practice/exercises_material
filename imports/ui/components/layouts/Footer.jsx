import React from "react";
import PropTypes from "prop-types";
import { Paper, Tabs, Tab } from "@material-ui/core";

const Footer = ({ muscles, category, onSelect }) => {
  const index = category ? muscles.findIndex(group => group === category) + 1 : 0;

  const onIndexSelect = (e, index) => {
    onSelect(index === 0 ? "" : muscles[index - 1]);
  };

  return (
    <Paper>
      <Tabs
        value={index}
        onChange={onIndexSelect}
        indicatorColor="primary"
        textColor="primary"
        centered
      >
        <Tab key="All" label="All" />
        {muscles.map(muscleGroup => <Tab key={muscleGroup} label={muscleGroup} />)}
      </Tabs>
    </Paper>
  );
};

Footer.propTypes = {
  muscles: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Footer;
