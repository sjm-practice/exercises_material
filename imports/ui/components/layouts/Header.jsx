import React from "react";
import { AppBar, Toolbar, Typography } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import CreateDialog from "../exercises/CreateDialog";

const styles = {
  flex: {
    flex: 1,
  },
};

const Header = ({ classes, muscles, onExerciseCreate }) => (
  <AppBar position="static">
    <Toolbar>
      <Typography variant="headline" color="inherit" className={classes.flex}>
        Exercise Database
      </Typography>
      <CreateDialog muscles={muscles} onCreate={onExerciseCreate} />
    </Toolbar>
  </AppBar>
);

export default withStyles(styles)(Header);
