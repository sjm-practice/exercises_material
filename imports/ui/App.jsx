/* eslint-disable arrow-parens, object-curly-newline react/destructuring-assignment */
import React, { Component, Fragment } from "react";
import { withStyles } from "@material-ui/core/styles";
import styles from "./styles";
import Header from "./components/layouts/Header";
import Exercises from "./components/exercises/Exercises";
import Footer from "./components/layouts/Footer";
import { muscles, exercises } from "../data/store";
import AlertDialogSlide, { alertDialog } from "./components/global/AlertDialogSlide";

// App component - represents the whole app
export default withStyles(styles)(
  class App extends Component {
    state = {
      exercises,
      exercise: {},
    };

    getExercisesByMuscle() {
      return muscles.map(group => {
        const { exercises: groupEx } = this.state;
        return [group, groupEx.filter(ex => ex.muscles === group)];
      });
    }

    handleCategorySelect = category => {
      this.setState({
        category,
      });
    };

    handleExerciseSelect = id => {
      this.setState(({ exercises: exs }) => ({
        exercise: exs.find(ex => ex.id === id),
      }));
    };

    handleExerciseCreate = exercise => {
      this.setState(({ exercises: exs }) => ({
        exercises: [...exs, exercise],
      }));
    };

    handleExerciseDelete = id => {
      this.setState(({ exercises: exs }) => ({
        exercises: exs.filter(ex => ex.id !== id),
      }));
    };

    handleExerciseEdit = id => {
      alertDialog({
        message: `editing an exercise: ${id}`,
        context: `Here is the context message. yo.`,
        button1: {
          label: `Buttton 1`,
          handler: () => alert("button one clicked."),
        },
        button2: {
          label: `Buttton 2`,
          handler: () => alert("button two clicked."),
        },
      });
    };

    render() {
      const { classes } = this.props; // eslint-disable-line no-unused-vars

      const exs = this.getExercisesByMuscle();
      const { category, exercise } = this.state;

      return (
        <Fragment>
          <Header muscles={muscles} onExerciseCreate={this.handleExerciseCreate} />
          <Exercises
            exercise={exercise}
            category={category}
            exercises={exs}
            onSelect={this.handleExerciseSelect}
            onDelete={this.handleExerciseDelete}
            onEdit={this.handleExerciseEdit}
          />
          <Footer
            category={category}
            muscles={muscles}
            onSelect={this.handleCategorySelect}
          />
          <AlertDialogSlide />
        </Fragment>
      );
    }
  },
);
