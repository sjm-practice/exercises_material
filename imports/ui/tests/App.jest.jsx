/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import App from "../App";
import Header from "../components/layouts/Header";
import Footer from "../components/layouts/Footer";

describe("<App />", function() {
  it("matches render snapshot", function() {
    const tree = renderer.create(<App />);
    expect(tree).toMatchSnapshot();
  });

  it("contains a Footer and Header components", function() {
    const wrapper = mount(<App />);
    expect(wrapper.find(Header)).toHaveLength(1);
    expect(wrapper.find(Footer)).toHaveLength(1);
  });
});
