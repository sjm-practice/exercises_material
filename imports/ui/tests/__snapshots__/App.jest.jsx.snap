// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches render snapshot 1`] = `
Array [
  <div>
    <h1>
      Header
    </h1>
  </div>,
  <div>
    <h1>
      Footer
    </h1>
  </div>,
]
`;
